### ISSUE TITLE
yt not loading all particles in Flash

### ISSUE THREAD

**Issue by jslavin** [yt-project/yt#4370]
### Bug report

**Bug summary**

yt doesn't load the data for all the particles in a Flash data file.

For two nearly identicle runs loading the data and displaying it shows all 10 particles in one run and only 9 for the other run. The only difference in the runs is that for the second the tracer particle data includes values for which particles_posy are non-zero whereas for the first all such values are zero. In the hdf5 files the blocks and all other data (for example as viewed with hdfview) match between the two files. The simulation was done in 1D.

```python
ds = yt.load(particles_data_file1)
ds.r['all','particles_tag']
# shows 10 values

ds2 = yt.load(particles_data_file2)
ds2.r['all','particles_tag']
# shows 9 values
```

**Version Information**

  * Operating System: Linux 4.18.0-425.3.1.el8.x86_64
  * Python Version: 3.8.16
  * yt version: 4.1.4
  * Other Libraries (if applicable):

python and yt were installed with conda

This could have something to do with bug #3681 but in this case all the blocks are present in the hdf5 file with positive block numbers.

**Comment by matthewturk** [comment]
Hi @jslavin sorry it's taken me some time to look at this, but I am optimistic I will be able to today or tomorrow.  Any chance you can provide a data file for me to look at, in case I can't replciate it here?

**Comment by jslavin** [comment]
Yes I have two files. Where would be a good place to upload them so that you can access them?

**Comment by matthewturk** [comment]
If they are smaller than a couple megs you can attach them here, otherwise
execute:

`curl -T file1.hdf5 https://use.yt/upload/`

and put the returned URL here.

**Comment by jslavin** [comment]
They're small but github wouldn't allow me to attach them because of not being one of the accepted file types. I'll upload to the site you list. The files are just the particle files. I don't think you need the associated hydro data files. The file names are nonzero_posy_hdf5_part_0024 and only_zero_posy_hdf5_part_0024

**Comment by jslavin** [comment]
Sorry, missed the part about the returned urls:
[http://use.yt/upload/45fd4e64](url)
[http://use.yt/upload/1a7337e5](url)

**Comment by jslavin** [comment]
Hey Matthew, have you been able to reproduce the bug?

**Comment by jslavin** [comment]
Hi Matthew,

It has something to do with the selector. By default the domain_left_edge gets values of 0 for y and z and domain_right_edge gets values of 1 for y and z. This is because the simulation was run in 1D, I think. However, I haven't been able to find where the selector is applied, though it seems to be after the dataset is instantiated. So it seems to be when I get the tags via:
tags = ds.r['all_data','particle_tags']
It seems to me that when one is requesting all_data, then having a RegionSelector operating is not right (if that is indeed what's happening). I tried altering the HDF5 file and setting ymin to -1 and ymax to 1 and when I do that, I get all the particles.

Jon

**Comment by matthewturk** [comment]
@jslavin I am so sorry for not seeing this before.  I think that you are right, we need to make sure that using a region selector isn't a problem, but I also think that may be some code paths that we turn into "Always True" for these.  Would you be able to diagram what you think is going on, and I'll work on addressing it?

**Comment by jslavin** [comment]
So I was able to do a run in 1D but with the limits on y made larger. With ymin and ymax set to values that cover the full range of yvalues in the particles, then the particles are not discareded by yt. So that is a workaround, though I have some long simulations that I'd hate to have to run again just to change that. Or to edit all those HDF5 files.
The goal of the simulation is to model a shock in 1D but the particles (dust grains) are allowed to move in (at least) 2D.
So, as I said before, I think what's happening is that yt is reading ymin and ymax in the HDF5 file and imposing those limits on the particles somehow. Perhaps there needs to be some additional option to prevent a region selector from being applied when one is not desired.
It's been a while since I looked at the yt code, so I can't really say much more than that right now.
